builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -257,7 +263,47 @@ func resourceAwsS3BucketDelete(d *schema.ResourceData, meta interface{}) error {
 		Bucket: aws.String(d.Id()),
 	})
 	if err != nil {
-		return err
+		ec2err, ok := err.(aws.APIError)
+		if ok && ec2err.Code == "BucketNotEmpty" {
+			if d.Get("force_destroy").(bool) {
+				// bucket may have things delete them
+				log.Printf("[DEBUG] S3 Bucket attempting to forceDestroy %+v", err)
+
+				bucket := d.Get("bucket").(string)
+				resp, err := s3conn.ListObjects(
```

Comment:
@m-s-austin does this have some upper bound of how many it returns? Do we need to paginate this?